### ISSUE TITLE
Linker error in xcode causing iOS build to fail after update

### ISSUE THREAD

**Issue by RoboticAccolade** [googleads/googleads-mobile-unity#1706]
### [REQUIRED] Step 1: Describe your environment

Unity version: 2019.2.20f1
Google Mobile Ads Unity plugin version: 6.0.1
Platform: iOS (iOS, Android, Unity Editor)
Platform OS version: iOS 9+ (eg iOS 10, Android 9)
Any specific devices issue occurs on: build level issue
Mediation ad networks used, and their versions: Facebook Ads Mediation v.3.3.0, Tapjoy v.12.7, AdColony v.4.1.4

#### Steps to reproduce:

What happened? How can we make the problem occur?
This could be a description, log/console output, etc.
 
I managed to resolve, at least as far as I can tell, my previous issue #1678 thanks to the advice given to replace the archive file with the unzipped folder from the Github source.  Unfortunately, I still can't successfully build the iOS version due to a linker error with Google Mobile Ads functions.

Let me say up front, there's a definite possibility that this error is due to some bad setting or old code hook in our project; it's several years old at this point and there are a number of plugins (plus some poorly removed plugins) in it that could be causing some sort of conflict either out of necessity or due to poor settings on my part.

In any case, the issue I'm getting currently is an error with the xcode linker (the Ld command): 
```
16:43:21 Undefined symbols for architecture armv7:
16:43:21   "_GADUMediationAdapterClassNameForRewardedAd", referenced from:
16:43:21       _Externs_GADUMediationAdapterClassNameForRewardedAd_m49EB4232988F403B8C4D3C67C2678D80E9F45B99 in Assembly-CSharp2.o
16:43:22      (maybe you meant: _Externs_GADUMediationAdapterClassNameForRewardedAd_m49EB4232988F403B8C4D3C67C2678D80E9F45B99)
16:43:22   "_GADURequestRewardedAd", referenced from:
16:43:22       _Externs_GADURequestRewardedAd_mFE6165AFAB704E23F5D3B63EF2FFD7F2C1D10513 in Assembly-CSharp2.o
16:43:22      (maybe you meant: _Externs_GADURequestRewardedAd_mFE6165AFAB704E23F5D3B63EF2FFD7F2C1D10513)
16:43:22   "_GADURewardedAdReady", referenced from:
16:43:22       _Externs_GADURewardedAdReady_mFFD83BF7B3152D5DFF5CB6DD348184ABC9F612C0 in Assembly-CSharp2.o
16:43:22      (maybe you meant: _Externs_GADURewardedAdReady_mFFD83BF7B3152D5DFF5CB6DD348184ABC9F612C0)

...

16:43:27 ld: symbol(s) not found for architecture armv7
16:43:27 clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

To be frank, I don't know enough about xcode linking and cocoapods to troubleshoot this problem.  Checking my build folder, I did notice the following:  the GoogleMobileAds framework in the pods folder in the xcode project is different between my successful build and the above error version:
<img width="439" alt="スクリーンショット 2021-07-19 10 18 34" src="https://user-images.githubusercontent.com/76253638/126090463-bcb5a921-9a19-4cc6-b851-4433d7c9636a.png">
This is what the framework looks like with the older mobile ads version installed, which will successfully build a working version.  In comparison here is what it looks like on my failed attempt to update:
<img width="439" alt="スクリーンショット 2021-07-19 9 53 28" src="https://user-images.githubusercontent.com/76253638/126090556-45c4b1dc-3ac0-4006-b6a9-0d6371061dba.png">
I don't know if this folder layout is just a result of pulling a higher version of the framework (and I'm not sure where the GoogleMobileAds framework version is declared in the first place) or if there is a setting somewhere that is causing it to pull both armv7 and arm64 versions - and I don't know if the presence of both versions is the reason the linker is failing.  Given the error message though, I suspect it may be related.

Obviously if anyone has encountered and solved this issue I'd love to hear about it - I'd also appreciate any advice that I could be given regarding why the pod might have 2 libraries in it and whether that could cause a problem.

For the record, my successful build's podfile entries:
PODS:
  - Google-Mobile-Ads-SDK (7.67.0):
    - GoogleAppMeasurement (~> 6.0)
    - GoogleUserMessagingPlatform (~> 1.1)
  - GoogleMobileAdsMediationTapjoy (12.7.0.0):
    - Google-Mobile-Ads-SDK (>= 7.65.0)
    - TapjoySDK (= 12.7.0)
DEPENDENCIES:
  - Google-Mobile-Ads-SDK (= 7.67)

And the failed update attempt:
PODS:
  - Google-Mobile-Ads-SDK (8.7.0):
    - GoogleAppMeasurement (< 9.0, >= 7.0)
    - GoogleUserMessagingPlatform (>= 1.1)
  - GoogleMobileAdsMediationTapjoy (12.7.0.0):
    - Google-Mobile-Ads-SDK (>= 7.65.0)
    - TapjoySDK (= 12.7.0)
DEPENDENCIES:
  - Google-Mobile-Ads-SDK (~> 8.2)

**Comment by NVentimiglia** [comment]
What version of XCode are you currently using?

**Comment by RoboticAccolade** [comment]
Currently our build machine is running XCode 12.4.1.  Since updating past that involves an OS update, we're holding off on that for now and avoiding using the archive file in 6.0.1.

**Comment by NVentimiglia** [comment]
Can you try building using SDK 6.0.2

If it persists, I will look into replicating on our end.

**Comment by Archi97** [comment]
> Currently our build machine is running XCode 12.4.1. Since updating past that involves an OS update, we're holding off on that for now and avoiding using the archive file in 6.0.1.

Update to 12.5.* and you will be able to build your project. 

@NVentimiglia if you will follow other issues, user already found the issue. its all on you now to fix the problems ;)

**Comment by NVentimiglia** [comment]
Fixed in 6.0.2
